internal/fetch_helper.go

```diff
@@ -172,10 +173,17 @@ func DownloadAndDecompressStorageFile(folder storage.Folder, fileName string) (i
 // TODO : unit tests
 // DownloadFileTo downloads a file and writes it to local file
 func DownloadFileTo(folder storage.Folder, fileName string, dstPath string) error {
+	file, err := os.OpenFile(dstPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC|os.O_EXCL, 0666)
+	if err != nil {
+		return err
+	}
+
 	reader, err := DownloadAndDecompressStorageFile(folder, fileName)
 	if err != nil {
 		return err
```

Comment:
if an error occurs, we will abandon an empty file - not really good, we should remove it